File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Order/Common/UserOrderListResponse.cs
using ByteBuy.Core.Domain.Enums;
using ByteBuy.Core.DTO.Public.Money;

namespace ByteBuy.Core.DTO.Public.Order.Common;

public record UserOrderListResponse(
    Guid OrderId,
    OrderStatus Status,
    DateTime PurchasedDate,
    string SellerDisplayName,
    int LinesCount,
    MoneyDto TotalItemsCost,
    MoneyDto DeliveryCost,
    MoneyDto TotalCost,
    IReadOnlyCollection<UserOrderLineResponse> Lines);

File: Byte&Buy.AngularSPA/src/app/core/dto/order/common/user-order-list-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../common/money-dto";
import { OrderStatus } from "../enum/order-status";
import { UserSaleOrderLineResponse } from "../sale/user-sale-order-line-response";
import { UserRentOrderLineResponse } from "../rent/user-rent-order-line-response";

export type UserOrderLineResponseUnion =
    | UserSaleOrderLineResponse
    | UserRentOrderLineResponse;

export interface UserOrderListResponse {
    orderId: Guid;
    status: OrderStatus;
    sellerDisplayName: string;
    buyerDisplayName: string;
    purchasedDate: Date;
    linesCount: number;
    isDeletable: boolean;
    totalItemsCost: MoneyDto;
    deliveryCost: MoneyDto;
    totalCost: MoneyDto;
    lines: UserOrderLineResponseUnion[];
}